File: Anul II/Medii de proiectare si programare/Laborator/lab08/C#/Ubb.BikeContest.Server/ProtocolBufferServer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Sockets;
using System.Text;
using System.Threading.Tasks;
using Ubb.BikeContest.Networking.ProtocolBuffers;
using Ubb.BikeContest.Services;

namespace Ubb.BikeContest.Server
{
    public class ProtocolBufferServer : ConcurrentServer
    {
        private readonly IContestServices _server;
        private ProtocolBufferWorker _worker;

        public ProtocolBufferServer(string host, int port, IContestServices server) : base(host, port)
        {
            _server = server;
        }

        protected override Thread CreateWorker(TcpClient client)
        {
            _worker = new ProtocolBufferWorker(_server, client);
            return new Thread(new ThreadStart(_worker.Run));
        }
    }
}

File: Anul II/Medii de proiectare si programare/Laborator/lab10/server/src/main/java/com/ubb/server/ProtocolBufferServer.java
package com.ubb.server;

import com.ubb.IContestServices;
import com.ubb.protocolbuffers.ProtocolBufferWorker;

import java.net.Socket;

public class ProtocolBufferServer extends ConcurrentServer {
    private final IContestServices server;

    public ProtocolBufferServer(int port, IContestServices server) {
        super(port);
        this.server = server;
    }

    @Override
    protected Thread createWorker(Socket client) {
        ProtocolBufferWorker worker = new ProtocolBufferWorker(server, client);
        return new Thread(worker);
    }
}
